benchopt/plotting/plot_from_csv.py

```diff
@@ -0,0 +1,150 @@
+
+for idx_data, dataset in enumerate(datasets):
+    df1 = df[df['data_name'] == dataset]
+    for idx_obj, objective in enumerate(objectives):
+        df2 = df1[df1['objective_name'] == objective]
+        ax = axarr[idx_data, idx_obj]
+        # check that at least one solver converged to compute c_star
+        # if df2["objective_duality_gap"].min() > FLOATING_PRECISION * df2["objective_duality_gap"].max():
+        #     print(
+        #         f"No solver reached a duality gap below {FLOATING_PRECISION}, "
+        #         "cannot safely evaluate minimum objective."
+        #     )
+        #     continue
+        c_star = np.min(df2["objective_value"]) - FLOATING_PRECISION
+        for i, solver_name in enumerate(solvers):
+            df3 = df2[df2['solver_name'] == solver_name]
+            curve = df3.groupby('stop_val').median()
+
+            y = curve["objective_value"] - c_star
+
+            ax.loglog(
+                curve["time"], y, color=CMAP(i), marker=MARKERS[i], markersize=6,
+                label=solver_name, linewidth=2)
+
+        ax.set_xlim([DICT_XLIM.get(dataset, MIN_XLIM), ax.get_xlim()[1]])
+        axarr[len(datasets)-1, idx_obj].set_xlabel(
+            "Time (s)", fontsize=fontsize - 2)
+        axarr[0, idx_obj].set_title(
+            '\n'.join(regex.search(objective).group(1).split(",")), fontsize=fontsize - 2)
+        ax.tick_params(axis='both', which='major', labelsize=labelsize)
+
+    if regex.search(dataset) is not None:
+        dataset_label = (regex.sub("", dataset) + '\n' +
+                         '\n'.join(regex.search(dataset).group(1).split(',')))
+    else:
+        dataset_label = dataset
+    axarr[idx_data, 0].set_ylabel(
+        dataset_label, fontsize=fontsize - 6)
+
+main_fig.suptitle(regex.sub('', objective), fontsize=fontsize)
+plt.show(block=False)
+
+# plot legend on separate fig
+leg_fig, ax2 = plt.subplots(1, 1, figsize=(20, 4))
+n_col = 3
+if n_col is None:
+    n_col = len(axarr[0, 0].lines)
+
+# take first ax, more likely to have all solvers converging
+ax = axarr[0, 0]
+lines_ordered = list(itertools.chain(*[ax.lines[i::n_col] for i in range(n_col)]))
+legend = ax2.legend(
+    lines_ordered, [line.get_label() for line in lines_ordered], ncol=n_col,
+    loc="upper center")
+leg_fig.canvas.draw()
+leg_fig.tight_layout()
+width = legend.get_window_extent().width
+height = legend.get_window_extent().height
+leg_fig.set_size_inches((width / 80,  max(height / 80, 0.5)))
+plt.axis('off')
+plt.show(block=False)
+
+
+if SAVEFIG:
+    Path('./figures').mkdir(exist_ok=True)
+    main_fig_name = f"figures/{figname}.pdf"
+    main_fig.savefig(main_fig_name)
+    os.system(f"pdfcrop {main_fig_name} {main_fig_name}")
+    main_fig.savefig(f"figures/{figname}.svg")
+
+    leg_fig_name = f"figures/{figname}_legend.pdf"
+    leg_fig.savefig(leg_fig_name)
+    os.system(f"pdfcrop {leg_fig_name} {leg_fig_name}")
+    leg_fig.savefig(f"figures/{figname}_legend.svg")
```

Comment:
```suggestion
    leg_fig.savefig(f"figures/{figname}_legend.svg", dpi=300)
```

Reply:
I like having the svg if I need to pimp the fig quickly before putting in a presentation of a rebuttal